File: app/src/main/java/com/originb/inkwisenote2/modules/textnote/data/TextNotesDao.java
package com.originb.inkwisenote2.modules.textnote.data;

import androidx.room.*;

import java.util.List;
import java.util.Set;

@Dao
public interface TextNotesDao {

    @Query("SELECT * FROM text_notes")
    List<TextNoteEntity> getAllTextNotes();

    @Query("SELECT * FROM text_notes WHERE book_id = :bookId")
    List<TextNoteEntity> getTextNoteForBook(Long bookId);

    @Query("SELECT * FROM text_notes WHERE note_id = :noteId")
    TextNoteEntity getTextNoteForNote(Long noteId);

    @Query("SELECT * FROM text_notes WHERE book_id IN (:bookIds)")
    List<TextNoteEntity> getTextNoteForBooks(Set<Long> bookIds);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insertTextNote(TextNoteEntity TextNoteEntity);

    @Update
    int updateTextNote(TextNoteEntity TextNoteEntity);

    @Update
    int updateTextNotes(List<TextNoteEntity> TextNoteEntities);

    @Query("DELETE FROM text_notes WHERE note_id = :noteId")
    void deleteTextNote(long noteId);

    @Query("DELETE FROM text_notes WHERE book_id = :bookId")
    void deleteTextNotes(long bookId);
}

File: app/src/main/java/com/originb/inkwisenote2/modules/textnote/data/TextNotesDao.kt
package com.originb.inkwisenote2.modules.textnote.data

import androidx.room.*

@Dao
interface TextNotesDao {
    @get:Query("SELECT * FROM text_notes")
    val allTextNotes: MutableList<TextNoteEntity>

    @Query("SELECT * FROM text_notes WHERE book_id = :bookId")
    fun getTextNoteForBook(bookId: Long): MutableList<TextNoteEntity>

    @Query("SELECT * FROM text_notes WHERE note_id = :noteId")
    fun getTextNoteForNote(noteId: Long): TextNoteEntity

    @Query("SELECT * FROM text_notes WHERE book_id IN (:bookIds)")
    fun getTextNoteForBooks(bookIds: MutableSet<Long>): MutableList<TextNoteEntity>

    @Insert(onConflict = OnConflictStrategy.Companion.REPLACE)
    fun insertTextNote(TextNoteEntity: TextNoteEntity): Long

    @Update
    fun updateTextNote(TextNoteEntity: TextNoteEntity): Int

    @Update
    fun updateTextNotes(TextNoteEntities: MutableList<TextNoteEntity>): Int

    @Query("DELETE FROM text_notes WHERE note_id = :noteId")
    fun deleteTextNote(noteId: Long)

    @Query("DELETE FROM text_notes WHERE book_id = :bookId")
    fun deleteTextNotes(bookId: Long)
}
